src/miner.cpp

```diff
@@ -120,6 +120,10 @@ bool IsValidMinerAddress(const MinerAddress& minerAddr) {
     return minerAddr.which() != 0;
 }
 
+bool IsShieldedMinerAddress(const MinerAddress& minerAddr) {
+    return minerAddr.which() == 1;
+}
```

Comment:
With `boost::variant`, yes. After #4892 has been merged, this will become:
```suggestion
bool IsShieldedMinerAddress(const MinerAddress& minerAddr) {
    return std::holds_alternative<libzcash::SaplingPaymentAddress>(minerAddr);
}
```
or alternatively (since we won't be adding new transparent miner address types):
```suggestion
bool IsShieldedMinerAddress(const MinerAddress& minerAddr) {
    return !(
        std::holds_alternative<InvalidMinerAddress>(minerAddr) ||
        std::holds_alternative<boost::shared_ptr<CReserveScript>>(minerAddr));
}
```